src/Dijkstra.NET/ShortestPath/ShortestPathExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Dijkstra.NET.Graph;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Dijkstra.NET.ShortestPath
+{
+    public static class ShortestPathExtensions
+    {
+        public static IEnumerable<TEdgeCustom> GetReversedPathEdgesCustom<T, TEdgeCustom>(this ShortestPathResult result, Graph<T, TEdgeCustom> graph) where TEdgeCustom : IEquatable<TEdgeCustom>
+        {
+            return result.GetPathEdgesCustom(graph).Reverse();
+        }
+
+        public static IEnumerable<TEdgeCustom> GetPathEdgesCustom<T, TEdgeCustom>(this ShortestPathResult result, Graph<T, TEdgeCustom> graph) where TEdgeCustom : IEquatable<TEdgeCustom>
+        {
+            if (result.IsFounded)
+            {              
+                using (var iterator = result.GetPath().GetEnumerator())
+                {
+                    if (!iterator.MoveNext())
+                    {
+                        yield break;
+                    }
+                    var currentNode = iterator.Current;
+                    while (iterator.MoveNext())
+                    {
+                        yield return GetFirstEdgeCustom(graph, currentNode, iterator.Current);
```

Comment:
What if between two nodes will bo more than one edge ?

Reply:
Hello. Mea culpa. I implemented this method to work with my custom aggregated info (GROUP BY from,to) so I have only one groupped edge between nodes in path. I will take a look what will be good realization of this veriant